src/classes/CSVReaderTest.cls

```diff
@@ -0,0 +1,151 @@
+/**
+* File Name :   CSVReaderTest.cls
+* Description : Made to only test the methods in CSVReader.cls
+*
+* Modification Log
+* ============================================================================
+* Ver Date       Author         Modification
+* --- ---------- -------------- --------------------------
+* 0.1 07/04/2014 Klea Kolaric   Initial version
+*
+*
+*/
+@isTest
+private class CSVReaderTest {
+
+    static testMethod void parseBasicCsvString()
+    {
+        Blob blobCSV = Blob.valueOf(basicCsvString);
+        
+        Integer numberOfRows = basicCsvString.split(CSVReader.lineTerminator).size();
+        Integer numberOfCols = basicCsvString.split(CSVReader.lineTerminator).get(0).split(',').size();
+        
+        Test.startTest();
+        List<List<String>> resultMatrix = CSVReader.readCSVFile(blobCSV);
+        Test.stopTest();
+        
+        system.assertEquals(resultMatrix.size(),numberOfRows);
+        for(List<String> rowList : resultMatrix)
+        {
+        	 system.assertEquals(rowList.size(),numberOfCols);
+        	 for(String strVal : rowList)
+        	 {
+        	 	system.assert(String.isNotBlank(strVal));
+        	 }
+        }
+    }
+    
+    static testMethod void parseQuotedCsvString()
+    {
+    	//add extra fields to the file that contains "" and replace them later ''
+    	String quotedString1 = '"I\'m a little teapot"';
+    	String quotedString2 = '"This is my handle"';
+    	String quotedString3 = 'Un! Deux! Trois! Dis: Miroir Noir!';
+    	
+        String quotedCsvString = basicCsvString;
+        quotedCsvString += quotedString1 + ',Row52' + CSVReader.lineTerminator;
+        quotedCsvString += 'Row61,' + quotedString2 + CSVReader.lineTerminator;
+        quotedCsvString += 'Row71,' + quotedString3 + CSVReader.lineTerminator;
+        
+        Integer numberOfRows = quotedCsvString.split(CSVReader.lineTerminator).size();
+        Integer numberOfCols = quotedCsvString.split(CSVReader.lineTerminator).get(0).split(',').size();
+        
+        Test.startTest();
+        List<List<String>> resultMatrix = CSVReader.readCSVFile(quotedCsvString);
+        Test.stopTest();
+        
+        system.assertEquals(resultMatrix.size(),numberOfRows);
+        for(List<String> rowList : resultMatrix)
+        {
+        	 system.assertEquals(rowList.size(),numberOfCols);
+        	 for(String strVal : rowList)
+        	 {
+        	 	system.assert(String.isNotBlank(strVal));
+        	 }
+        }
+        system.assertEquals(quotedString1.replace('"',''),resultMatrix.get(numberOfRows-3).get(0));
+        system.assertEquals(quotedString2.replace('"',''),resultMatrix.get(numberOfRows-2).get(1));
+        system.assertEquals(quotedString3.replace('"',''),resultMatrix.get(numberOfRows-1).get(1));
+    }
+    
+    static testMethod void parseCsvStringWithEmptyLines()
```

Comment:
Style issue: test methods should be annotated with @IsTest for better readibility